### PR TITLE
Use JUnit5 `@ParameterizedTest`

### PR DESCRIPTION
**Pull request by JackPGreen**
JUnit 5's `@ParameterizedTest` annotation is much more flexible as it allows parameters to be specified per-test (rather than per-class), and is simpler.

The test results (in Eclipse at least) are easier to read as well:
JUnit4:
Screenshot 2023-10-12 at 09 15 39

JUnit5:
Screenshot 2023-10-12 at 09 15 11

I've migrated a couple of tests over, primarily to introduce the `junit-jupiter-params` artifact in a dedicated PR to be then used in subsequent development.

### CODE REVIEW

**Review by orcunc — CHANGES_REQUESTED** [review]
It seems the JUnit4 and JUnit5 imports/annotations are mixed within the same test. Can you fix that too because we should not do that

**Comment by JackPGreen** [comment]
> It seems the JUnit4 and JUnit5 imports/annotations are mixed within the same test. Can you fix that too because we should not do that

Good spot

47411f5681145b7deeefaaad5c16aaa88eda5d86

**Review by vbekiaris — APPROVED** [review]
nice cleanup

File: hazelcast/src/test/java/com/hazelcast/test/starter/test/HazelcastVersionLocatorTest.java [hunk 1]
```diff
@@ -19,64 +19,54 @@
...
-import org.junit.BeforeClass;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized.Parameter;
-import org.junit.runners.Parameterized.Parameters;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 import org.junit.runners.Parameterized.UseParametersRunnerFactory;
```
Comment (orcunc) [inline hazelcast/src/test/java/com/hazelcast/test/starter/test/HazelcastVersionLocatorTest.java:28]: I think there is one JUnit4 and JUnit5 mix here

Comment (JackPGreen) [reply hazelcast/src/test/java/com/hazelcast/test/starter/test/HazelcastVersionLocatorTest.java]: 23474563f241fc4939b6278d771574dcaa96019b

**Review by orcunc — APPROVED** [review]
Thanks for the upgrade to jupiter params

**Pull request merged**